apps/showcase/src/components/training/training.component.ts

```diff
@@ -0,0 +1,238 @@
+      }
+      const sanitizedPath = resource.path.replace(new RegExp('^[.]/?'), '');
+      const parsedPath = sanitizedPath.split('/');
+      parsedPath.reduce((pointer, path, index) => {
+        if (path.indexOf('.') > -1) {
+          const parsedContent = JSON.parse(resource.content);
+          pointer[path] = {file: {contents: parsedContent[path].file.contents}} as FileNode;
+          return pointer;
+        } else if (index === parsedPath.length - 1) {
+          // eslint-disable-next-line no-prototype-builtins
+          if (pointer[path]?.hasOwnProperty('file')) {
+            throw new Error('Cannot override a file with a folder');
+          }
+          pointer[path] ||= {directory: {}};
+          const pathDirectory = (pointer[path] as DirectoryNode).directory;
+          const overrides = Object.entries(JSON.parse(resource.content) as {[key: string]: FileNode | DirectoryNode});
+          overrides.forEach(([key, content]) =>
+            // eslint-disable-next-line no-prototype-builtins
+            pathDirectory[key] = pathDirectory[key]?.hasOwnProperty('directory') ? {...(pathDirectory[key] as DirectoryNode).directory, ...content} : content
+          );
+          return pathDirectory;
+        } else {
+          pointer[path] = pointer[path] && Object.hasOwn(pointer[path], 'directory') ? pointer[path] : {directory: {}};
+          return (pointer[path] as DirectoryNode).directory;
+        }
+      }, fileSystemTree);
+      return fileSystemTree;
+    }, {} as FileSystemTree);
+  }
+
+  private async updateStepDynamicContent(step: TrainingStep, urls: { url: string; path: string }[], solutionProject = false) {
+    const resourcesPromise: Promise<Resource[]> = Promise.all(urls.map(
+      async ({path, url}) => ({
+        path,
+        content: await this.loadResource(url)
+      })
+    ));
+    const resources = await resourcesPromise;
+    const filesContent = this.getFilesContent(resources);
+    step.dynamicContent[solutionProject ? 'solutionProject' : 'project'].set({
+      startingFile: step.description.filesConfiguration!.startingFile || '',
+      commands: step.description.filesConfiguration!.commands || [],
+      files: filesContent,
+      cwd: step.description.filesConfiguration!.name.toLowerCase().replace(' ', '-') + (solutionProject ? '-solution' : '')
+    });
+  }
+
+  public ngOnInit() {
+    void this.loadSteps();
+  }
+
+  public async loadResource(url: string) {
+    const resourceUrl = url.startsWith('./') ? `${this.trainingPath}/${url}` : `${url}`;
+    const response = await fetch(await firstValueFrom(this.dynamicContentService.getMediaPathStream(resourceUrl)));
+    return (await response.text());
+  }
+
+  public setCurrentStep(index: number) {
+    if (index > this.steps().length || index < 0) {
+      return;
+    }
+    this.currentStepIndex = index;
+    let newHash;
+    if (location.hash.match(/#([0-9]+)$/)) {
```

Comment:
regexp could be extracted in a `const` outside the component as it's used several time
and the name of the `const` will provide some documentation about it